_opensearch/index-templates.md

```diff
@@ -184,19 +175,233 @@ PUT _index_template/template-02
+
+```json
+PUT _index_template/daily_logs
+{
+  "index_patterns": [
+    "logs-2020-01-*"
+  ],
+  "template": {
+    "aliases": {
+      "my_logs": {}
+    },
+    "settings": {
+      "number_of_shards": 2,
+      "number_of_replicas": 1
+    },
+    "mappings": {
+      "properties": {
+        "timestamp": {
+          "type": "date",
+          "format": "yyyy-MM-dd HH:mm:ss||yyyy-MM-dd||epoch_millis"
+        },
+        "value": {
+          "type": "double"
+        }
+      }
+    }
+  },
+  "priority": 200,
+  "composed_of": [
+    "component_template_1",
+    "component_template_2"
+  ],
+  "version": 3,
+  "_meta": {
+    "description": "using component templates"
+  }
+}
+```
+
+If you create an index named `logs-2020-01-01`, you can see that it derives it’s mappings and settings from both the component templates:
+
+```json
+PUT logs-2020-01-01
+GET logs-2020-01-01
+```
+
+#### Sample response
+
+```json
+{
+  "logs-2020-01-01": {
+    "aliases": {
+      "my_logs": {}
+    },
+    "mappings": {
+      "properties": {
+        "@timestamp": {
+          "type": "date"
+        },
+        "ip_address": {
+          "type": "ip"
+        },
+        "timestamp": {
+          "type": "date",
+          "format": "yyyy-MM-dd HH:mm:ss||yyyy-MM-dd||epoch_millis"
+        },
+        "value": {
+          "type": "double"
+        }
+      }
+    },
+    "settings": {
+      "index": {
+        "creation_date": "1625382479459",
+        "number_of_shards": "2",
+        "number_of_replicas": "1",
+        "uuid": "rYUlpOXDSUSuZifQLPfa5A",
+        "version": {
+          "created": "7100299"
+        },
+        "provided_name": "logs-2020-01-01"
+      }
+    }
+  }
+}
+```
+
+### Simulate multi-component templates
+
+For index templates composed of multiple component templates, you can simulate applying a new template to verify whether the settings are applied as you expect.
+
+To simulate the settings that would be applied to a specific index name:
+
+```json
+POST _index_template/_simulate_index/<index_name>
+```
+
+To simulate the settings that would be applied from an existing template:
+
+```json
+POST _index_template/_simulate/<index_template>
+```
+
+You can also specify a template definition in the simulate request:
+
+```json
+POST _index_template/_simulate
+{
+  "index_patterns": [
+    "logs-2020-01-*"
+  ],
+  "template": {
+    "settings" : {
+        "index.number_of_shards" : 3
+    }
+  },
+  "composed_of": ["component_template_1", "component_template_2"]
+}
+```
+
+The `_simulate` API returns the final settings, mappings, and aliases that will be applied to indices that match the index pattern. You can also see any overlapping templates whose configuration is superseded by the simulated template body or higher-priority templates:
```

Comment:
I don't think we need the hyphen on higher priority.

Reply:
Fixed.